pandas/core/reshape/pivot.py

```diff
@@ -427,6 +427,9 @@ def _convert_by(by):
 @Substitution("\ndata : DataFrame")
 @Appender(_shared_docs["pivot"], indents=1)
 def pivot(data: "DataFrame", index=None, columns=None, values=None) -> "DataFrame":
+    if columns is None:
+        raise ValueError("pivot() missing 1 required argument: 'columns'")
```

Comment:
This should be a TypeError no? I think @simonjayhawkins had that original comment

Reply:
sorry, i thought `ValueError` is more appropriate here for this case.

I changed to `TypeError` as suggested by @simonjayhawkins ! thanks both!

Reply:
> This should be a TypeError no? I think @simonjayhawkins had that original comment

I _think_ it should be TypeError, however, I would regard a change from ValueError to TypeError as a breaking change. Not sure what others think.

Reply:
nvm. The exception in the issue is a KeyError, so either is a change.

Reply:
get it! thanks!